### PR TITLE
Fixes underterministic collision detection ordering in IdMapper

### PR DESCRIPTION
**Pull request by tinwelint**
since the sorting of the collision data didn't take into account each
items position in dataCache. Now it does.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/idmapping/string/EncodingIdMapper.java [hunk 1]
```diff
@@ -452,9 +450,50 @@ private void buildCollisionInfo( InputIterator<Object> ids, ProgressListener pro
...
+            }
+
+            @Override
+            public boolean ge( long right, long pivot )
+            {
+                long rightEId = dataCache.get( right );
+                long pivotEId = dataCache.get( pivot );
+                if ( DEFAULT.ge( rightEId, pivotEId ) )
+                {
+                    return rightEId == pivotEId ? right > pivot : true;
```
Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/idmapping/string/EncodingIdMapper.java:487]: `right >= pivot`?

**Comment by davidegrohmann** [comment]
@tinwelint It would be nice to have a test for the original problem that this PR is aiming to fix.

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/idmapping/string/EncodingIdMapper.java (hunk 1 above)
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/idmapping/string/EncodingIdMapper.java:487]: correct, although there's no chance that they're ever the same actually. They just can't be. It should be an assertion if anything.

**Comment by tinwelint** [comment]
There is a test which failed at least 50% of the times before. And the issue (ordering) is random since ParallelSort pivot selection is random in the qsort algorithm, so it'd be quite hard to test unless there were refactorings in these areas to expose this functionality in isolation so that it could be tested. I don't think it's a good time to do that since we're dealing with a flaky test. But going forward we could refactor in small steps. It's slightly scary since this code is quite complicated.

**Pull request merged**